tronbyt_server/manager.py

```diff
@@ -1137,3 +1137,149 @@ def moveapp(device_id: str, iname: str) -> ResponseReturnValue:
+
+    # Convert the device dictionary to JSON
+    device_json = json.dumps(device, indent=4)
+
+    # Create a response to serve the JSON as a file download
+    response = Response(
+        device_json,
+        mimetype="application/json",
+        headers={"Content-Disposition": f"attachment;filename={device_id}_config.json"},
+    )
+    return response
+
+
+@bp.route("/<string:device_id>/update_config", methods=["GET", "POST"])
+@login_required
+def import_device_config(device_id: str) -> ResponseReturnValue:
+    if not validate_device_id(device_id):
+        abort(HTTPStatus.BAD_REQUEST, description="Invalid device ID")
+
+    if request.method == "POST":
+        # Check if the POST request has the file part
+        if "file" not in request.files:
+            flash("No file part")
+            return redirect(
+                url_for("manager.import_device_config", device_id=device_id)
+            )
+
+        file = request.files["file"]
+
+        # If no file is selected
+        if file.filename == "":
+            flash("No selected file")
+            return redirect(
+                url_for("manager.import_device_config", device_id=device_id)
+            )
+
+        # Ensure the uploaded file is a JSON file
+        if not file.filename.endswith(".json"):
+            flash("Invalid file type. Please upload a JSON file.")
+            return redirect(
+                url_for("manager.import_device_config", device_id=device_id)
+            )
+
+        try:
+            # Parse the JSON file
+            device_config = json.load(file)
+
+            # Validate the JSON structure (basic validation)
+            if not isinstance(device_config, dict):
+                flash("Invalid JSON structure")
+                return redirect(
+                    url_for("manager.import_device_config", device_id=device_id)
+                )
+
+            # Check if the device already exists
+            user = g.user
+            if (
+                device_config["id"] not in user["devices"]
+                and device_config["id"] == device_id
+            ):
+                flash("Not the same device id. Import skipped.")
+                return redirect(url_for("manager.index"))
+
+            # Add the new device to the user's devices
+            user["devices"][device_config["id"]] = device_config
+            db.save_user(user)
+
+            flash("Device configuration imported successfully")
+            return redirect(url_for("manager.index"))
+
+        except json.JSONDecodeError as e:
+            flash(f"Error parsing JSON file: {e}")
+            return redirect(
+                url_for("manager.import_device_config", device_id=device_id)
+            )
+
+    # Render the import form
+    return render_template("manager/import_config.html", device_id=device_id)
+
+
+@bp.route("/import_device", methods=["GET", "POST"])
+@login_required
+def import_device() -> ResponseReturnValue:
```

Comment:
`import_device` and `import_device_config` (which maybe should be `update_device_config` to match the URL) are almost identical. Maybe you could extract the shared logic into a function with a that indicates whether the given device ID must already exist or not.